core/tee/tee_rpmb_fs.c

```diff
@@ -571,7 +571,7 @@ static TEE_Result tee_rpmb_req_pack(struct rpmb_req *req,
 
 		if (rawdata->blk_idx) {
 			/* Check the block index is within range. */
-			if ((*rawdata->blk_idx + nbr_frms) >
+			if ((*rawdata->blk_idx + nbr_frms - 1) >
```

Comment:
Why should it fail on last block?
`max_blk_idx` seems the inclusive max index allowed value.

Reply:
I should not, but the code before this patch would fail on the last block I think, that's the problem I'm trying to fix.

Say the partition has 1024 blocks then `max_block_idx == 1023`. If I want to write one block of data at the last position, I set `blk_idx == 1023` and `nbr_frms == 1`. The test is therefore `1023 + 1 - 1 > 1023` which is false, add one to `blk_idx` and it becomes true.

One thing I don't understand is why this has not caused `xtest` failures. It seems the RPMB FS does not use the last block and I don't understand why, an allocator bug with `TEE_MM_POOL_HI_ALLOC` perhaps. For example running `xtest 6001` I see data are written to block index `1022`, not `1023`, so even with the "wrong" test all is good. That's a different issue IMO.

Reply:
oh, right. `nbr_frms - 1`.